src/main/java/edu/hw4/Queries.java

```diff
@@ -0,0 +1,151 @@
+package edu.hw4;
+
+import edu.hw4.Animal.Sex;
+import edu.hw4.Animal.Type;
+import edu.hw4.errors.AgeError;
+import edu.hw4.errors.HeightError;
+import edu.hw4.errors.NameError;
+import edu.hw4.errors.ValidationError;
+import edu.hw4.errors.WeightError;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class Queries {
+    private final List<Animal> animals;
+    private final static int ELEVEN_TASK_HEIGHT_LIMIT = 100;
+
+    public Queries(List<Animal> animals) {
+        this.animals = animals;
+    }
+
+    public List<Animal> one() {
+        return animals.stream().sorted(Comparator.comparingInt(Animal::height)).toList();
+    }
+
+    public List<Animal> two(int k) {
+        return animals.stream().sorted((a, b) -> Integer.compare(b.weight(), a.weight())).limit(k).toList();
```

Comment:
Было бы круто каждый новый метод стрима писать с новой строчки, а то так они расплываются в глазах.
```suggestion
        return animals.stream()
            .sorted((a, b) -> Integer.compare(b.weight(), a.weight()))
            .limit(k)
            .toList();
```